src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -502,6 +502,10 @@ private void validateSchema(final EventTypeBase eventType) throws InvalidEventTy
             final JSONObject schemaAsJson = new JSONObject(eventTypeSchema);
             final Schema schema = SchemaLoader.load(schemaAsJson);
 
+            if (schemaAsJson.has("type") && !schemaAsJson.getString("type").equals("object")) {
```

Comment:
There are several problems here: 
1. It makes no sense in previous call to SchemaLoader.load(schemaAsJson) if schema type is not "object", so it's better to move this call down
2. In case if "type" exists and is null, then there will be JSONException triggered, probably it is not what was expected, it is better to write something like `schemaAsJson.has("type") && Objects.equals("object", schemaAsJson.getString("type"))`

Reply:
@ferbncode